### PR TITLE
Unlock static build support for a bunch of providers

### PR DESCRIPTION
**Pull request by nirvn**
Static build support for the following providers:
- delimited text
- spatialite
- virtual layer
- wfs
- wcs
- afs
- ams

This also includes fix for progresql-less static builds.

Let the iOS consumers rejoice.

### CODE REVIEW

**Comment by nirvn** [comment]
@3nids , did a quick core-only static build, and the delimited text provider built just fine :partying_face:

**Comment by nirvn** [comment]
Unrelated external storage test failure (which we could consider disabling, seems to fail 1/10th of the time).

File: src/providers/virtual/CMakeLists.txt [hunk 1]
```diff
@@ -2,79 +2,117 @@
...
+
+  # require c++17
+  target_compile_features(provider_virtuallayer PRIVATE cxx_std_17)
+
+  target_compile_definitions(provider_virtuallayer PRIVATE "-DQT_NO_FOREACH")
+  
+  target_link_libraries(provider_virtuallayer
+    qgis_core
+    ${QT_VERSION_BASE}::Core
+    ${QT_VERSION_BASE}::Widgets
```
Comment (3nids) [inline src/providers/virtual/CMakeLists.txt:101]: do we need to link these 2? Linking qgis_core should be enough?

**Review by 3nids — APPROVED** [review]
awesome!

**Pull request merged**